### PR TITLE
Issue 1466

### PR DESCRIPTION
**Pull request by mbaechler**
#### Check List:
* Implements #1466 
* Unit tests : OF COURSE
* Javadoc with a code example (on API only) : YES

### CODE REVIEW

**Comment by mbaechler** [comment]
Anybody willing to review and merge this PR?

**Comment by joel-costigliola** [comment]
Has fallen under our radar, we'll try for the next release

File: src/test/java/org/assertj/core/api/iterable/IterableAssert_elements_Test.java [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+  void should_pass_allowing_assertions_for_repeating_elements() {
+  // WHEN
+    IterableAssert<String> result = assertThat(iterable).elements(2, 1, 2);
+    // THEN
+    result.containsExactly("Lisa", "Marge", "Lisa");
+  }
+
+  @Test
+  void should_fail_if_index_out_of_range() {
+    AssertionError assertionError = expectAssertionError(() -> assertThat(iterable).elements(5));
```
Comment (joel-costigliola) [inline src/test/java/org/assertj/core/api/iterable/IterableAssert_elements_Test.java:78]: minor: add `// WHEN`

**Review by joel-costigliola — CHANGES_REQUESTED** [review]

**Comment by mbaechler** [comment]
I rebased on `main` and added a fixup commit with fixes from your review.

**Comment by joel-costigliola** [comment]
Integrated thanks @mbaechler!
I made the code compatible with soft assertions which lead me to revert the changes you made due to one of my comments on using IllegalArgumentException for checking the indices validity (sorry I did see that).

**Pull request closed without merging**